File: Java/src/Main/Cake.java
package Main;

public abstract class Cake extends BakedGood {
    private final String flavor, frosting;

    public Cake(double basePrice, String flavor, String frosting) {
        super(basePrice);
        this.flavor = flavor;
        this.frosting = frosting;
    }

    public String getFlavor() {
        return flavor;
    }

    public String getFrosting() {
        return frosting;
    }
}

File: Python3/Cake.py
import BakedGood

class Cake(BakedGood):
    def __init__(self, basePrice, flavor, frosting):
        super().__init__(basePrice)
        self._flavor = flavor
        self._frosting = frosting

    def getFlavor(self):
        return self._flavor

    def getFrosting(self):
        return self._frosting
